addons/stock/models/stock_move.py

```diff
@@ -619,7 +619,7 @@ def _set_lot_ids(self):
                         move_line = mls_without_lots[:1]
                         move_lines_commands.append(Command.update(move_line.id, {
                             'lot_id': lot.id,
-                            'product_uom_id': move.product_id.uom_id.id,
+                            'product_uom_id': move.product_uom.id,
```

Comment:
only if the move is not a serial. You don't want to use pack of 6 on a move line with tracking by serial number